accepted/00000-hub-reporting.md

```diff
@@ -0,0 +1,177 @@
+The goal is now to get data from these Servers back and have combined reporting data available on the Hub.
+
+The data should be made available for external Reporting Tools.
+
+If possible, the data should also be usable on a single Uyuni Server with the same mechanism.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The data should be made available in a PostgreSQL database. We considered using other database types,
+but the increased maintenance effort for new packages and limited knowledge about NoSQL databases
+let us stay with the good well known PostgreSQL database.
+Also PostgreSQL seems to be accepted in most Reporting tools as data source, while NoSQL databases
+are not supported, or only via standard DBMS modules. Native support was rarely available.
+
+### The Database
+The main database is a PostgresDB in the Uyuni Hub system (but in the future it will be possible to 
+use also an external db): it stores all the information collected from all the server, and eventually
+aggregates them. Other databases with the same schema are also present in all the server, to collect 
+information for that system.
+All the databases (in the Uyuni Hub and in all Uyuni Server) needs to be made available on the network to either connect in a secure way (using the SSL
+certificates provided by Uyuni Server) with the reporting tool or (for Uyuni Server) with the Hub to gather the data. 
+If possible the connection to the main Uyuni DB from the outside should be forbidden.
+In summary:
+- DB in Uyuni Hub stores, collects and eventually aggregates data coming from all the DB in Uyuni Server,
+- DB in Uyuni Server stores it own data,
+- The reporting tool can be connected either to the Hub or to any Uyuni Server,
+- all connections are secure.
+
+### The Database Schema
+
+The schema should export the most important tables from the main Uyuni Database as a slightly de-normalized
+variant containing only data which are relevant for a report.
+
+A ready-to-use report can be provided with views doing joins over multiple tables.
+No foreign keys should be used to make data update and refreshing easier and independent from the order
+of the tables.
+
+Every table gets an extra column for the Uyuni server id of server which provide the data (`mgm_id`).
+On a single Uyuni Server this column has the standard value `1` which represent "localhost".
+On the Hub it will be replaced with the real server id the managed server has in the hub database.
+
+Every row should get a report timestamp column (`exported`), which is set to the time when the data
+were exported from the main Uyuni Server database.
+
+Indexes should be set on the typical columns which might be searched for.
+
+Data which belong to the traditional stack only (like osad status) should not be made available in the
+reporting database as the traditional stack will be deprecated soon.
+
+A detailed Database schema is not part of this RFC.
```

Comment:
I wholeheartedly agree producing a full schema for the RFC is not appropriate, but would it help to go already into the exercise of listing what are the main tables that we expect (and roughly how many they are), or at least what objects/entities/concepts we intend to cover.

Could a shortlist be added here?

Reply:
@mackdk maybe you can add what you prepared as an example.

Reply:
I added the example in the RFC. I don't think however that this fully answers @moio questions, as the example was build by examining only the inventory report and trying to generalise it. In order to have a better overview on what are the tables/entites/concepts we want to cover we need to scope a bit more the data and what pieces of information are valuable for the user.